server/dump.go

```diff
@@ -90,3 +98,52 @@ func (p *Plugin) createDump(ctx context.Context, min, max time.Time, remoteStora
 
 	return zipFileNameRemote, nil
 }
+
+// Metadata is the auxiliary data added to a plugin downloadable.
+// It should contain the required fields and may have it's custom values as well.
+// TODO: move this to mattermost/server/public/pluginapi in the future
+type metadata struct {
+	ServerVersion string         `yaml:"server_version"`
+	ServerID      string         `yaml:"server_id"`
+	LicenseID     string         `yaml:"license_id"`
+	PluginID      string         `yaml:"plugin_id"`
+	Custom        map[string]any `yaml:"custom"`
+}
+
+func (p *Plugin) genarateMetadataForDump(path string, min, max time.Time) (*metadata, error) {
+	m := &metadata{}
+	if l := p.API.GetLicense(); l != nil {
```

Comment:
TODO: let's make sure the values obtained here align to the similar ones collected in places like `support_packet.go` (we want to have the same license value across all support packets collected from the same customer, etc.).

Reply:
I'll double check that, in fact for future versions, we can ask these from server or even let server generate the metadata on behalf of the plugin.